soroban-env-host/src/auth.rs

```diff
@@ -835,30 +894,21 @@ impl AuthorizationTracker {
     // Tries to match the provided invocation to the authorized sub-invocation
     // of the current tree and push it to the call stack.
     // Returns `true` if the match has been found.
-    fn maybe_push_matching_invocation_frame(
-        &mut self,
-        contract_id: &Hash,
-        function_name: &ScSymbol,
-        args: &ScVec,
-    ) -> bool {
+    fn maybe_push_matching_invocation_frame(&mut self, function: &AuthorizedFunction) -> bool {
         let mut frame_index = None;
         if let Some(curr_invocation) = self.last_authorized_invocation_mut() {
             for i in 0..curr_invocation.sub_invocations.len() {
                 let sub_invocation = &mut curr_invocation.sub_invocations[i];
-                if !sub_invocation.is_exhausted
-                    && &sub_invocation.contract_id == contract_id
-                    && &sub_invocation.function_name == function_name
-                    && &sub_invocation.args == args
-                {
+                // TODO: the equality comparison should be metered.
+                if !sub_invocation.is_exhausted && &sub_invocation.function == function {
                     frame_index = Some(i);
                     sub_invocation.is_exhausted = true;
                     break;
                 }
             }
         } else if !self.root_authorized_invocation.is_exhausted
-            && &self.root_authorized_invocation.contract_id == contract_id
-            && &self.root_authorized_invocation.function_name == function_name
-            && &self.root_authorized_invocation.args == args
+            // TODO: the equality comparison should be metered.
```

Comment:
Can this TODO be resolved in this PR? If not, we need to open an issue for this.

Reply:
We have https://github.com/stellar/rs-soroban-env/issues/668 for all the auth metering stuff. Here I'm just really not sure how to do the right thing - unlike other XDRs that I've added in this PR, this one has dynamic size, so I'm not sure if the tools will do the right thing (maybe we need to implement `xdr_size` in rust if we want to address this more holistically...).